src/engine.ml

```diff
@@ -912,8 +919,8 @@ let expected_packet session transport data =
 type error =
   [ Packet.error
   | Lzo.error
-  | `Non_monotonic_replay_id of transport * Packet.header
-  | `Non_monotonic_sequence_number of transport * int32 option * Packet.header
+  | `Non_monotonic_replay_id of int32 * int32
+  | `Non_monotonic_sequence_number of int32 * int32 option
```

Comment:
This could be without an option because this error only makes sense when the sequence number is present. We need to use something else than `opt_guard` then.
```suggestion
  | `Non_monotonic_sequence_number of int32 * int32
```

Reply:
indeed true, but this requires some further changes to construct the value then.